cluster-autoscaler/processors/provreq/processor.go

```diff
@@ -106,7 +107,24 @@ func (p *provReqProcessor) Process(provReqs []*provreqwrapper.ProvisioningReques
 			continue
 		}
 	}
+	p.DeleteOldProvReqs(provReqs)
 }
 
 // Cleanup cleans up internal state.
-func (p *provReqProcessor) CleanUp() {}
+func (p *provReqProcessor) Cleanup() {}
+
+// DeleteOldProvReqs delete ProvReq that have terminal state (Provisioned/Failed == True) more than a week.
+func (p *provReqProcessor) DeleteOldProvReqs(provReqs []*provreqwrapper.ProvisioningRequest) {
+	for _, provReq := range provReqs {
+		conditions := provReq.Status.Conditions
+		provisioned := apimeta.FindStatusCondition(conditions, v1beta1.Provisioned)
```

Comment:
shouldn't we have booking expired here?
Can you add a test case for it as well, I think in the test case we only have one for old-failed

Reply:
Currently that doesn't make any difference to check BookingExpired condition or Provisioned, however in general I think every ProvReq should have Provisioned condition, while BookingExpired is not necessary, so I decided to use Provisioned condition. 
Do you have a strong opinion to use BookingExpired here?

Reply:
No, just curious why this one was used